From 6968345893178d2750b8872055498d2a6010a861 Mon Sep 17 00:00:00 2001
From: HATAYAMA Daisuke <d.hatayama@fujitsu.com>
Date: Wed, 8 Dec 2021 12:07:34 +0000
Subject: [PATCH 3/3] defs.h: fix breakage of compatibility of struct
 symbol_table_data for extension modules

Commit <2fab8fbc0c4f> ("symbols: Implement install and remove operations
for mod_symname_hash") added new member variable mod_symname_hash in the
middle of struct symbol_table_date, which breaks compatibility of struct
symbol_table_data for extension modules. As the result, crash trace command
results in segmentation fault.

Fixes: 2fab8fbc0c4f ("symbols: Implement install and remove operations for mod_symname_hash")
Signed-off-by: HATAYAMA Daisuke <d.hatayama@fujitsu.com>
---
 defs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/defs.h b/defs.h
index 7d3ed78fcd23..b63741c7d78b 100644
--- a/defs.h
+++ b/defs.h
@@ -2753,7 +2753,6 @@ struct symbol_table_data {
         double val_hash_searches;
         double val_hash_iterations;
         struct syment *symname_hash[SYMNAME_HASH];
-	struct syment *mod_symname_hash[SYMNAME_HASH];
 	struct symbol_namespace kernel_namespace;
 	struct syment *ext_module_symtable;
 	struct syment *ext_module_symend;
@@ -2780,6 +2779,7 @@ struct symbol_table_data {
 	ulong kaiser_init_vmlinux;
 	int kernel_symbol_type;
 	ulong linux_banner_vmlinux;
+	struct syment *mod_symname_hash[SYMNAME_HASH];
 };
 
 /* flags for st */
-- 
2.20.1

